src/app/api/v1/preimages/route.ts

```
@@ -0,0 +1,67 @@
+import MESSAGES from '@/global/messages';
+import { PREIMAGE_LISTING_LIMIT } from '@/global/constants/listingLimit';
+import { API_ERROR_CODE } from '@/global/constants/errorCodes';
+import { urqlClient } from '@/services/urqlClient';
+import { GET_PREIMAGES, GET_STATUS_HISTORY_BY_PREIMAGES_HASH } from '@/app/api/v1/subsquidQueries';
+import { IPreimageResponse, IPreimagesListingResponse, IPreimageStatusHistory } from '@/global/types';
+import getReqBody from '../../api-utils/getReqBody';
+
+export const POST = withErrorHandling(async (req: NextRequest) => {
+	const { page = 1, hash = '' } = await getReqBody(req);
+
+	if (!page || isNaN(page) || Number(page) < 1) throw new APIError(`${MESSAGES.REQ_BODY_ERROR}`, 500, API_ERROR_CODE.REQ_BODY_ERROR);
```

Biome

[error] isNaN is unsafe. It attempts a type coercion. Use Number.isNaN instead.

See the MDN documentation for more details.
Unsafe fix: Use Number.isNaN instead.


(lint/suspicious/noGlobalIsNan)